### DISCUSSION TITLE
[QUESTION] Is pnpx deprecated?

### DISCUSSION THREAD

**Discussion by angelhdzmultimedia** [pnpm/pnpm#6477] [Q&A] [answered]
I've read that pnpm dlx replaces pnpx, but I always execute pnpx without issues. I just read that pnpx was deprecated.
Is this true and if so, why?

**Comment by zkochan** [comment, marked as answer]
As of now, `pnpx` is still present when you install pnpm through npm (and probably through some other installation methods). But it is not present if you install pnpm using the standalone script.

Overall, `pnpm dlx` is not a frequently used command, so I personally think it is not worth the effort to ship an additional binary. You can always create aliases if you want to type less. `pnpx` used to be more frequently used when it was also used for executing local binaries. However, now you should execute local binaries with `pnpm <binary name>`.

**Reply by angelhdzmultimedia** [reply]
> As of now, `pnpx` is still present when you install pnpm through npm (and probably through some other installation methods). But it is not present if you install pnpm using the standalone script.
> 
> Overall, `pnpm dlx` is not a frequently used command, so I personally think it is not worth the effort to ship an additional binary. You can always create aliases if you want to type less. `pnpx` used to be more frequently used when it was also used for executing local binaries. However, now you should execute local binaries with `pnpm <binary name>`.

Thanks for the answer. I asked because I work with Vue, Nuxt and NestJS, and I find myself executing a lot of code with "npx" so I immediately convert it to pnpx and it works. For example, `pnpm nuxi init nuxt-app` doesn't work. Only `pnpx nuxi init nuxt-app` and `pnpm dlx init nuxt-app` work.

Also `pnpx prisma generate` and `pnpx prisma init` for Prisma.

So in this case, should I keep using `pnpx`, or is `pnpm dlx` the recommended?
